src/rc_string.c

```diff
@@ -108,11 +109,61 @@ IMPLEMENT_MOCKABLE_FUNCTION(, THANDLE(RC_STRING), rc_string_create, const char*,
 }
 
 
+
 THANDLE(RC_STRING) rc_string_create_with_format(const char* format, ...)
 {
-    (void)format;
+    THANDLE(RC_STRING) result = NULL;
 
-    return NULL;
+    if (format == NULL)
+    {
+        /*Codes_SRS_RC_STRING_07_001: [ If format is NULL, rc_string_create_with_format shall fail and return NULL. ]*/
+        LogError("Invalid arguments: const char* format=%s", MU_P_OR_NULL(format));
+    }
+    else
+    {
+        /*Codes_SRS_RC_STRING_07_002: [ Otherwise, `rc_string_create_with_format` shall determine the total number of characters written using the variable number of arguments. ]*/
+        va_list args;
+        va_start(args, format);
+        size_t string_length_with_terminator = vsnprintf(NULL, 0, format, args) + 1;
```

Comment:
>vsnprintf [](http://example.com/codeflow?start=47&length=9)

So, what if vsnprintf return -2?
Section 7.19.6.12 of C99 says this can happen.

The vsnprintf function returns the number of characters that would have been written
had n been sufficiently large, not counting the terminating null character, *oraneg ative
value if an encoding error occurred*. 

Would like to see a test for it :-)